server/src/zmq_worker.rs

```diff
@@ -304,38 +304,41 @@ async fn handle_incoming_request(
                         .context("ZmqWorker error while forwarding request to load balancer.")
                 }
                 _ => {
-                    error!(
-                        "ZmqWorker received a request with api {:?} while I can only handle \
-                    Status/Metadatas/PtPlanner/PlacesNearby/NextDepartures/NextArrivals api.",
-                        requested_api
-                    );
-                    Ok(())
+                    let err = format_err!("ZmqWorker received a request with api {requested_api:?} while I can only handle \
+                    Status/Metadatas/PtPlanner/PlacesNearby/NextDepartures/NextArrivals api.");
+                    error!("{:?}", err);
+                    // let's send back a response to our zmq client that we received an invalid query
+                    send_error_response(zmq_socket, request_message.client_id, &err).await
                 }
             }
         }
         Err(err) => {
-            let err_str = format!("Could not decode zmq message into protobuf. {:?}", err);
-            error!("{}", err_str);
-
+            let err = format_err!(err).context("Could not decode zmq message into protobuf.");
```

Comment:
I'm not sure why you need `format_err!()` here. From what I understand, `err` is a [`prost::DecodeError`](https://docs.rs/prost/0.11.6/prost/struct.DecodeError.html) which implements [`core::error::Error`](core::error::Error). I didn't test, but you should be able to call `.context()` directly on `err`? :thinking: 

```suggestion
            let err = err.context("Could not decode zmq message into protobuf.");
```

Reply:
I don't know why, but the [Context](https://docs.rs/anyhow/1.0.69/anyhow/trait.Context.html) trait is not implemented for `T : impl core::error::Error` (but it is for `T : Result<T,E> `or `T : Option<U>` :shrug: ).
So i cannot call `context() `directly on `err`.
The `format_err!()`  [creates](https://docs.rs/anyhow/1.0.69/anyhow/macro.format_err.html) an `anyhow::Error` from a generic `core::error::Error`